base_multi_company/models/base.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Tecnativa - Pedro M. Baeza
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models
+
+
+class Base(models.AbstractModel):
+    _inherit = "base"
+
+    def _check_company(self, fnames=None):
+        """Inject as context the company of the record that is going to be compared
+        for being taking into account when computing the company of many2one's
+        relations that links with our multi-company models.
+        """
+        if self._name == "res.company":
+            company_source_id = self.id
+        else:
+            company_source_id = self.company_id.id
```

Comment:
`base` model inherits all the models itself but not all the models have `company_id` field or yes?

You can reproduce the error using `self.env["res.lang"].company_id`
 - `AttributeError: 'res.lang' object has no attribute 'company_id'`


```suggestion
        company_source_id = False
        if self._name == "res.company":
            company_source_id = self.id
        elif hasattr(self, "company_id"):
            company_source_id = self.company_id.id
```

Notice the following odoo-bin shell commands:
 - <img width="1155" alt="Screenshot 2023-11-28 at 3 54 03 p m" src="https://github.com/OCA/multi-company/assets/6644187/a60b1162-0a96-4c6c-8618-7d3635a115eb">

Reply:
`elif "company_id" in self._fields:`

Reply:
AFAIK, this method is only run when `_check_company_auto` is set to True in the model from `create` and `write`, which, by definition, requires a `company_id` field.

You can try on runboat to write any record in `res.lang`, and you will see that no error is raised. I have also tried changing lang on a res.partner.

Do you have any use case where the error is triggered?

Reply:
@pedrobaeza 
You have a good point!

I don't have a functional case triggering this error

I'm only seeing a future technical error for corner cases where we don't know if we should protect

Before this PR you can call: `self.env["res.lang"].search([])._check_company()` without errors

After this PR this will raise error

But you have a good point, maybe there are not code calling the method directly like this


e.g.
 - ![Screenshot 2023-11-28 at 5 48 40 p m](https://github.com/OCA/multi-company/assets/6644187/f69c1a09-0781-4a98-ac3d-57d94fafba0b)

Reply:
OK, no problem. Let's include the safeguard. It's barely overhead, so it's not so important.